wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt

```diff
@@ -971,6 +991,32 @@ class SwiftGenerator private constructor(
     )
   }
 
+  private fun TypeSpec.Builder.generateMessageDefaults(
+    type: MessageType,
+  ) {
+    val hasDefaults = type.fields.any { it.default != null }
+    if (hasDefaults) {
+      addType(
+        TypeSpec.enumBuilder("Defaults")
+          .addModifiers(PUBLIC)
+          .apply {
+            type.fields.forEach { field ->
+              val default = field.default
+              if (default != null) {
+                val defaultValue = defaultFieldInitializer(field.type!!, default)
```

Comment:
Checking for when I integrate this:

If I prune an `enum` with a raw value of `0`, what happens?

I'm _guessing_ I'm going to have to stop pruning the various `DO_NOT_USE` enum values, which kind of sucks but 🤷‍♂️

Reply:
I'd like to try your example? Is there a way to force this to happen? We can try it offline in our repo if you'd like.